### ISSUE TITLE
EFS throughput mode

### ISSUE THREAD

**Issue by troydieter** [neiman-marcus/terraform-aws-jenkins-ha-agents#17]
Hi-

Have you ever experienced bursting issues in EFS with Jenkins workplaces living there? I've opened an issue on [another repo](https://github.com/widdix/aws-cf-templates/issues/324) from what i've seen in the past. 

It may be worthy to add the following to the `aws_efs_file_system` resource in main.tf:

`throughput_mode = [var.efs_mode]`
`provisioned_throughput_in_mibps = [var.efs_provisioned_count]`

and then add the corresponding variable options in variables.tf

**Comment by claydanford** [comment]
Hello!

Thank you for commenting.  We have not experienced such issues, but it is a good idea to allow for this as an option, or perhaps default. I am investigating some options now. I'm thinking something like the following should work:

```HCL
resource "aws_efs_file_system" "master_efs" {
  ...
  throughput_mode                 = var.efs_mode
  provisioned_throughput_in_mibps = var.efs_mode == "provisioned" ? var.efs_provisioned_throughput : null
}

variable "efs_module" {
  default     = "bursting"
  description = "The EFS throughput mode. Options are bursting and provisioned. To set the provisioned throughput in mibps, configure efs_provisioned_throughput variable."
}

variable "efs_provisioned_throughput" {
  default     = 1
  description = "The EFS provisioned throughput in mibps. Ignored if efs throughput mode is set to bursting."
}
```

What are your thoughts?

**Comment by troydieter** [comment]
Hi,

Looks good - I'd just increase the default `efs_provisioned_throughput` to at least 3 (in the past, I've deployed it with 5.. which is about $35/mo for the EFS provisioned option).

**Comment by claydanford** [comment]
Resolved with v2.2.0. Thank you!
